### PR TITLE
Add session property for MetadataQueryOptimizer

### PR DESCRIPTION
**Pull request by zhenxiao**
https://github.com/facebook/presto/issues/4297

### CODE REVIEW

**Comment by zhenxiao** [comment]
@martint @electrum @dain @cberner kindly ping

File: presto-main/src/main/java/com/facebook/presto/SystemSessionProperties.java [hunk 1]
```diff
@@ -180,6 +181,11 @@ public SystemSessionProperties(
                         DICTIONARY_AGGREGATION,
                         "Enable optimization for aggregations on dictionaries",
                         featuresConfig.isDictionaryAggregation(),
+                        false),
+                booleanSessionProperty(
+                        OPTIMIZE_METADATA_QUERY,
+                        "Enalbe optimization for metadata queries",
```
Comment (electrum) [inline presto-main/src/main/java/com/facebook/presto/SystemSessionProperties.java]: Typo "Enalbe"

File: presto-main/src/main/java/com/facebook/presto/SystemSessionProperties.java [hunk 2]
```diff
@@ -54,6 +54,7 @@
     public static final String COLUMNAR_PROCESSING = "columnar_processing";
     public static final String COLUMNAR_PROCESSING_DICTIONARY = "columnar_processing_dictionary";
     public static final String DICTIONARY_AGGREGATION = "dictionary_aggregation";
+    public static final String OPTIMIZE_METADATA_QUERY = "optimize_metadata_query";
```
Comment (electrum) [inline presto-main/src/main/java/com/facebook/presto/SystemSessionProperties.java]: Should this be plural (and match the config property)?

**Comment by zhenxiao** [comment]
thank u, @electrum , comments addressed

**Comment by martint** [comment]
Merged, thanks!

**Pull request closed without merging**